modules/pool/lib/pool-on-chain-data.service.ts

```diff
@@ -101,13 +103,19 @@ export class PoolOnChainDataService {
             this.options.vaultAddress,
             networkContext.chain === 'ZKEVM' ? 190 : 1024,
         );
+        const tokenPairData = await fetchTokenPairData(
+            filteredPools,
+            this.options.balancerQueriesAddress,
+            networkContext.chain === 'ZKEVM' ? 190 : 1024,
```

Comment:
can you set chain from internal getter? it will help to pass them as config params later.